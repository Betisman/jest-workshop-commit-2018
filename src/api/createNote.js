
export default async (title, description) => {
  const response = await fetch('http://localhost:3001/note', {
    method: 'post',
    body: JSON.stringify({ title, description }),
    headers: { 'Content-Type': 'application/json' },
  });
  const notes = await response.json();
  return notes;
};