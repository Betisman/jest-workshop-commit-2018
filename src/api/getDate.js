
export default async () => {
  const response = await fetch('http://worldclockapi.com/api/json/est/now');
  const date = await response.json();
  return date;
};
