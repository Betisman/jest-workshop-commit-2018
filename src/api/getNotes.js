
export default async () => {
  const response = await fetch('http://localhost:3001/notes');
  const notes = await response.json();
  return notes;
};
