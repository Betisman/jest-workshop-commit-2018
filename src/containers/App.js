import React from 'react';
import Header from '../components/Header';
import Button from '../components/Button';

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      route: 'home',
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState({ route: 'app' });
  }

  render() {
    return (
      <div>
        <Header>
          Create your cool notes
        </Header>
        <main>
          <Button onClick={this.handleClick}>Access</Button>
        </main>
      </div>
    );
  }
}

export default App;
