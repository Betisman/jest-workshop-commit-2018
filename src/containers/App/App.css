
.appContainer {
  display: flex;
  width: 100vw;
  height: 80vh;
  justify-content: center;
  align-items: center;
}

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 25px;
}
