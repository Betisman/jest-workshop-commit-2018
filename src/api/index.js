import getDate from './getDate';
import createNote from './createNote';
import getNotes from './getNotes';

export {
  getDate,
  createNote,
  getNotes,
};
