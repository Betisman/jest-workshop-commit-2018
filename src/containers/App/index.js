import { connect } from 'react-redux';
import { getNotes, createNote } from '../../actions';
import App from './App';

const mapStateToProps = (state) => ({
  notes: state.notes,
});

const mapDispatchToProps = (dispatch) => ({
  getNotes: () => dispatch(getNotes()),
  createNote: (title, description) => dispatch(createNote(title, description)),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
