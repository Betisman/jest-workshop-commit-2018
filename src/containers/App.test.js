import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';
import { date } from '../../test/fixtures/api';

jest.mock('../api/getDate');

const props = {};

describe('<App />', () => {
  
  const shallowWrapper = shallow(<App />);
  const mountWrapper = mount(<App />);

  it('<Button /> match snapshot', () => {
    const tree = renderer.create(<App {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  console.log(shallowWrapper.debug());
  console.log(mountWrapper.debug());

  it('must render Header component' , () => {
    expect(shallowWrapper.find('Header').exists()).toEqual(true);
  });

  it('must render Button component' , () => {
    expect(shallowWrapper.find('Button').exists()).toEqual(true);
  });

  it('must have state property route with "home" value', () => {
    expect(shallowWrapper.state().route).toEqual('home');
  });

  it('Button click change state property route to "app" value', () => {
    shallowWrapper.find('Button').props().onClick();
    expect(shallowWrapper.state().route).toEqual('app');
  });

  it.only('must have state property date not {}', async () => {
    const newShallowComponent = await shallow(<App />);
    const result = newShallowComponent.state().date;
    expect(result).toEqual(date);
  });
});

