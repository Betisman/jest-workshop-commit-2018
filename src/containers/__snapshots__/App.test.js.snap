// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> <Button /> match snapshot 1`] = `
<div>
  <header>
    <h1
      className="ui header undefined"
    >
      Create your cool notes
    </h1>
  </header>
  <main>
    <button
      onClick={[Function]}
    >
      Access
    </button>
  </main>
</div>
`;
